### PR TITLE
Upgrade to Joi 13

### PR DESCRIPTION
**Pull request by stevewillard**
I took a stab at updating Joi to the latest version, since Hapi 17 relies on Joi 13

The only annoying thing is that you can no longer destructure the different Joi types from a require:
```
const { string } = require('joi')
```
Which means you need `Joi.x` everywhere. It doesn't look like there's a way around it, and the author makes it sounds like it's not going to be changed.

See https://github.com/hapijs/joi/issues/1349 for more details

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -46,7 +44,7 @@ module.exports = (schema) => {
     if (validationResult.error) {
       if (testMode) {
         const variableValidationError = validationResult.error.details.find(
-          detail => detail.path === variable
+          detail => detail.path.includes(variable)
```
Comment (joegoldbeck) [inline index.js]: Is this change due to a change in the Joi api?

Comment (stevewillard) [reply index.js]: I think so? I didn't see it as part of the change log in the last few major revisions, but path is returning as an array of strings now.

Comment (joegoldbeck) [reply index.js]: Hmmm, since it's so small, maybe we write it so it works in either scenario and we can maintain cross-version compatibility? What do you think?

Comment (stevewillard) [reply index.js]: Sure, that sounds reasonable

**Comment by stevewillard** [comment]
Ok I think we're good. Hopefully the `peerDependency` is correct. The tests are still passing at least:
```
$ yarn test
yarn run v1.3.2
$ NODE_ENV=test mocha --recursive --timeout 1000


  tsEnv
    ✓ accepts a Joi schema object
    ✓ returns a function "getEnv"
    getEnv
      ✓ throws an error if setEnv hasnt been called with an object
      ✓ accepts a string
      ✓ returns the value of the environment variable from process.env, properly coerced
      ✓ returns the value of optional environment variables
      ✓ throws if requested environment variable is not defined in schema
      ✓ throws if any variable in process.env cannot be coerced to the correct type
      ✓ strips unknown environment variables
      ✓ throws if process.env is missing environment variables in schema
      ✓ does not throw if optional environment variables in schema are not in process.env
      ✓ throws if forbidden environment variables in schema are in process.env
      ✓ does not obey default values set in the schema
      ✓ requires process.env.NODE_ENV to equal 'test', 'local', 'demo', 'development', 'staging', 'production'
      ✓ requires process.env.SERVICE_NAME to be a non-empty string
      ✓ requires process.env.TENANT to be a non-empty string
      in test mode
        ✓ returns the value of environment variable which have been set, properly coerced
        ✓ throws if value of requested environment variable has not been set
        ✓ throws if requested environment variable is not defined in schema
        ✓ throws if requested variable in process.env cannot be coerced to the correct type
        ✓ strips unknown environment variables
        ✓ resets the environment to {} if no argument passed


  22 passing (36ms)

✨  Done in 0.41s.
```

File: index.js [hunk 2]
```diff
@@ -44,7 +44,7 @@ module.exports = (schema) => {
     if (validationResult.error) {
       if (testMode) {
         const variableValidationError = validationResult.error.details.find(
-          detail => detail.path.includes(variable)
+          detail => detail.path === variable || detail.path.includes(variable)
```
Comment (joegoldbeck) [inline index.js]: Won't we get `details.path.include is not a function` if detail.path is a string?

Comment (joegoldbeck) [inline index.js]: We should probably have an array check here, otherwise there is a weird edge case.

`'my_var_1'.includes('my_var')` === `true`

Comment (stevewillard) [reply index.js]: Oh I think you're right. I was thinking it would short circuit, but if they're not equal, it'll explode. I'll re-test with version 10

Comment (joegoldbeck) [reply index.js]: It might not explode actually since strings have a .include method (I updated my comment). Rather, there is a weird edge case.

'my_var_1'.includes('my_var') === true
